### DISCUSSION TITLE
How to set output of the action inside a docker container?

### DISCUSSION THREAD

**Discussion by jonyhy96** [community/community#25154] [Actions]
<p>Problem when building an action with docker, i want to set some outputs of my action but i can’t find some document about how to set output of an action inside a container.</p>
<p>By just send some message to stderr/stdout ? Is there some format of output that i should follow?</p>
<p>Please give some help! Thank you!</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jonyhy96">@jonyhy96</a> ,</p>
<p>Thank you for being here!</p>
<p>You can refer to the doc <a href="https://help.github.com/en/actions/building-actions/creating-a-docker-container-action#create-an-action-metadata-file" rel="nofollow noopener">here</a> to check the docker action example for the output setting.</p>
<ol>
<li>
<p>In the action.yaml datameta file, define output:</p>
<p>outputs:<br>
time: # id of output<br>
description: ‘The time we greeted you’</p>
</li>
<li>
<p>In the entrypoint.sh file,  you must use a workflow command in a specific syntax: echo “::set-output name=&lt;output name&gt;::&lt;value&gt;”</p>
<p>#!/bin/sh -l</p>
<p>time=$(date)<br>
echo “::set-output name=time::$time”</p>
</li>
</ol>
<p>Hope it helps!</p>

**Comment by jonyhy96** [comment]
<p>Thanks for help!</p>
<p>I already know that we can use <code>echo "::set-output name=\&lt;output name\&gt;::\&lt;value\&gt;"</code> inside an entrypoint.sh to declare an output.</p>
<p>I wanna to know what if i don’t use entrypoint.sh but use a binary(written in golang) to handle the input. How can i set the output of the action by the binary? By execute a shell command(::set-output name=&lt;output name&gt;::&lt;value&gt;) inside my binary? What is the principle of ::set-output ?</p>

**Comment by ethomson** [comment, marked as answer]
<p>&gt; By execute a shell command(::set-output name=&lt;output name&gt;::&lt;value&gt;) inside my binary?</p>
<p>No, that’s not a shell command - that’s the parameter to <code>echo</code>.  By printing that string to stdout, the GitHub Actions runner will store that name/value pair in the output context for that step.  eg, in Go:</p>
<pre><code>import("fmt")

fmt.Println("::set-output name=output_name::value")
</code></pre>
<p>&gt; By just send some message to stderr/stdout ?</p>
<p>Yes, send these commands to stdout.</p>
<p>&gt; Is there some format of output that i should follow?</p>
<p>Yes, they’re documented here: <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-commands-for-github-actions</a></p>

**Reply by marlinhares** [reply]
This is not the best answer anymore.
As in 03/2022 The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Reply by rr-jino-jose** [reply]
any example how to set this in env files way?

**Reply by gipo355** [reply]
hey @rr-jino-jose did you find a solution?

I'm experiencing exactly this. A go cli program inside a container action must write to outputs

**Reply by jonahw23** [reply]
Try something like this @gipo355 
```
githubEnv := os.Getenv("GITHUB_ENV")
if githubEnv != "" {
        f, err := os.OpenFile(githubEnv, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
        if err != nil {
            fmt.Println("Error opening GITHUB_ENV file:", err)
            return
        }
        defer f.Close()
        if _, err := f.WriteString(fmt.Sprintf("{var name}=%s\n", {value})); err != nil {
            fmt.Println("Error writing to GITHUB_ENV file:", err)
            return
        }
    }

**Reply by gipo355** [reply]
Indeed GitHub mounts the volumes and we have access to GITHUB_OUTPUT variable which points to the file that must be appended to to add the outputs
